restore-sql-server-transaction-logs/Function/main.py

```diff
@@ -0,0 +1,274 @@
+
+    while log_file_processed == False:
+
+        request_attempts += 1
+
+        try:
+            
+            logger.info(f"Attempt to process object {object_name}. Try {request_attempts} out of {max_request_attempts}...")
+
+            request = service.instances().import_(project=project_id, instance=csql_instance_name, body=import_context_body)
+            response = request.execute()
+            
+            logger.info(f"Executed request: {import_context_body}")
+                        
+            operation_status_done = False
+            operation_status_fetch_attempts=0
+
+            while not operation_status_done:
+
+                operation_status_fetch_attempts += 1
+                operation_sleep_time_seconds = min(max_operation_fetch_time_seconds, 5*operation_status_fetch_attempts)
+                
+                logger.info(f"Waiting for {operation_sleep_time_seconds} seconds to check for the operation status...")
+                
+                sleep(operation_sleep_time_seconds)
+                
+                resp = service.operations().get(project=project_id, operation=response["name"]).execute()
+
+                logger.info(f"Fetched operation status: {resp['status']}")
+
+                if resp["status"] == "DONE":
+                    
+                    operation_status_done = True                    
+
+                    if resp.get("error") is not None:
+                        
+                        outcome_message, finished_state = handle_error(resp.get("error"), source_bucket, object_name, logger)
+
+                        if finished_state:
+                            return outcome_message, 200
+                        else:
+                            raise RuntimeError(outcome_message)
+
+                    else:
+
+                        #operation success
+                        logger.info(f"IMPORT DONE. Response details: {resp}")
+                        log_file_processed=True
+                        
+                        copy_blob_to_processed_bucket(processed_bucket_name, storage_client, source_bucket, object_name, logger)
+                        delete_processed_blob(source_bucket, object_name, logger)
+                        
+                        return "Operation succeded.", 200
+
+        except Exception as err:
+            logger.info(f"Request error: {err.args}")
+            request_sleep_time_seconds = min(max_request_fetch_time_seconds, 5*request_attempts)
+            logger.info(f"Another operation might be in progress. Retrying after {request_sleep_time_seconds} seconds...")
+            sleep(request_sleep_time_seconds)
+
+        finally:
+            if request_attempts >= max_request_attempts:
+                logger.info(f"Number of retries ({request_attempts}) reached. Exiting function.")
+                return "Operation abandoned.", 500

```

Comment:
What happens then? Is the event that triggered the execution lost?

Refer to how [Eventarc retries events](https://cloud.google.com/eventarc/docs/retry-events).

Reply:
Indeed, now that I think about it, the Eventarc retries make sense.

Reply:
Done.

Reply:
LGTM